category_encoders/utils.py

```diff
@@ -175,6 +182,162 @@ def get_generated_cols(X_original, X_transformed, to_transform):
     return current_cols
 
 
+class BaseEncoder(BaseEstimator):
+    _dim: Optional[int]
+    cols: List[str]
+    use_default_cols: bool
+    handle_missing: str
+    handle_unknown: str
+    verbose: int
+    drop_invariant: bool
+    invariant_cols: List[str] = []
+    feature_names: List[str] = []
+    return_df: bool
+    supervised: bool
+
+    def fit(self, X, y=None, **kwargs):
+
+        # first check the type
+        if self.supervised:
+            X, y = convert_inputs(X, y)
+        else:
+            X = convert_input(X)
+
+        self._dim = X.shape[1]
+
+        # if columns aren't passed, just use every string column
+        # todo this is a construct from LOO that applies to all encoders. However we should look for a proper way to make encoders re-fitable
+        if self.use_default_cols:
+            self.cols = get_obj_cols(X)
+        else:
+            self.cols = convert_cols_to_list(self.cols)
+
+        if self.handle_missing == 'error':
+            if X[self.cols].isnull().any().any():
+                raise ValueError('Columns to be encoded can not contain null')
+
+        self._fit(X, y, **kwargs)
+
+        # if self.supervised:
+        #     X_temp = self.transform(X, y, override_return_df=True)
+        # else:
+        X_temp = self.transform(X, override_return_df=True)  # for finding invariant columns transform as if it was the test set
+        self.feature_names = X_temp.columns.tolist()
+
+        # drop all output columns with 0 variance.
+        if self.drop_invariant:
+            generated_cols = get_generated_cols(X, X_temp, self.cols)
+            self.invariant_cols = [x for x in generated_cols if X_temp[x].var() <= 10e-5]
+            try:
+                [self.feature_names.remove(x) for x in self.invariant_cols]
```

Comment:
```suggestion
                for x in self.invariant_cols:
                    self.feature_names.remove(x)
```

don't use a list comp for its side effects